File: src/main/java/com/manson/fo76/repository/UserRepository.java
package com.manson.fo76.repository;

import com.manson.fo76.domain.dto.User;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends MongoRepository<User, String> {

  User findByName(String name);
}

File: src/main/java/com/manson/fo76/repository/UserRepository.kt
package com.manson.fo76.repository

import com.manson.fo76.domain.User
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Service

@Service
interface UserRepository : MongoRepository<User?, String?> {
    fun findByName(name: String?): User?
    fun deleteByName(name: String?): User?
}